src/codewhisperer/util/codewhispererStates.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export class CodeWhispererStates {
+    static #instance: CodeWhispererStates
+
+    // Some other variables for client component latency
+    fetchCredentialStartTime = 0
+    sdkApiCallStartTime = 0
+    invokeSuggestionStartTime = 0
+
+    public static get instance() {
+        return (this.#instance ??= new CodeWhispererStates())
+    }
+
+    setFetchCredentialStart() {
+        if (this.fetchCredentialStartTime === 0 && this.invokeSuggestionStartTime !== 0) {
+            this.fetchCredentialStartTime = performance.now()
```

Comment:
`performance` is not available in node 14 (test in cloud9-codecatalyst)

https://github.com/aws/aws-toolkit-vscode/blob/a5031452fb1333429a6dcf26da19492e432c2612/src/codewhisperer/commands/startSecurityScan.ts#L36

Reply:
Fixing